tests/unit/test_observer.py

```diff
@@ -11,30 +11,43 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-
-from typing import List
-
-import numpy.testing as npt
 import numpy as np
 import pytest
 import tensorflow as tf
 
+from trieste.data import Dataset
 from trieste.observer import map_is_finite, filter_finite
 
+from tests.util.misc import ShapeLike, assert_datasets_allclose
+
 
-def _nan_at_origin(t: tf.Tensor) -> tf.Tensor:
+def _sum_with_nan_at_origin(t: tf.Tensor) -> tf.Tensor:
     is_at_origin = tf.reduce_all(t == [[0., 0.]], axis=-1, keepdims=True)
     sums = tf.reduce_sum(t, axis=-1, keepdims=True)
-    return tf.where(is_at_origin, tf.constant([[np.nan]]), sums)
+    return tf.where(is_at_origin, [[np.nan]], sums)
 
 
-def test_filter_finite() -> None:
-    ok_query_points = [[-1., 0.], [1., 0.], [0., 2.], [1., 3.]]
-    query_points = tf.constant([[0., 0.]] + ok_query_points)
-    finite_values = filter_finite(query_points, _nan_at_origin(query_points))
-
-    npt.assert_array_almost_equal(finite_values.query_points, ok_query_points)
-    npt.assert_array_almost_equal(finite_values.observations, [[-1.], [1.], [2.], [4.]])
+@pytest.mark.parametrize('query_points, expected', [
+    (  # one failure point
+        tf.constant([[-1., 0.], [1., 0.], [0., 2.], [0., 0.], [1., 3.]]),
+        Dataset(
+            tf.constant([[-1., 0.], [1., 0.], [0., 2.], [1., 3.]]),
+            tf.constant([[-1.], [1.], [2.], [4.]])
+        )
+    ),
+    (  # no failure points
+        tf.constant([[-1., 0.], [1., 0.], [0., 2.], [1., 3.]]),
+        Dataset(
+            tf.constant([[-1., 0.], [1., 0.], [0., 2.], [1., 3.]]),
+            tf.constant([[-1.], [1.], [2.], [4.]])
+        )
+    ),
+    (tf.constant([[0., 0.]]), Dataset(tf.zeros([0, 2]), tf.zeros([0, 1]))),  # only failure points
+    (tf.zeros([0, 2]), Dataset(tf.zeros([0, 2]), tf.zeros([0, 1]))),  # empty data
+])
+def test_filter_finite(query_points: tf.Tensor, expected: Dataset) -> None:
+    actual = filter_finite(query_points, _sum_with_nan_at_origin(query_points))
```

Comment:
I feel extracting a variable for `_sum_with_nan_at_origin(query_points)`, named `observations` would make this easier to read.

Reply:
will do